scalafix-core/shared/src/main/scala/scalafix/config/CustomMessage.scala

```diff
@@ -0,0 +1,44 @@
+package scalafix
+package config
+
+import metaconfig.{Conf, ConfError, ConfDecoder, Configured, Metaconfig}
+import org.langmeta._
+import scalafix.internal.config.MetaconfigPendingUpstream.XtensionConfScalafix
+
+import scala.language.implicitConversions
+
+class CustomMessage[T](value: T, message: Option[String], id: Option[String])
+
+object CustomMessage {
+  def getOption[T](obj: Conf.Obj, path: String)(
+      implicit ev: ConfDecoder[T]): Configured[Option[T]] =
+    Metaconfig
+      .getKey(obj, Seq(path))
+      .map(
+        value => ev.read(value).map(Some(_))
+      )
+      .getOrElse(Configured.Ok(None))
+
+  def decoder[T](field: String)(
+      implicit ev: ConfDecoder[T]): ConfDecoder[CustomMessage[T]] =
+    ConfDecoder.instance[CustomMessage[T]] {
+      case obj: Conf.Obj => {
+        (obj.get[T](field) |@|
+          getOption[String](obj, "message") |@|
```

Comment:
https://github.com/olafurpg/metaconfig/pull/31

replace with `obj.getOption[String]("message")`

Reply:
metaconfig v0.5.4 is out with `getOption` 🔥